src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/InfiniteScrollList.xaml.cs

```diff
@@ -86,31 +86,63 @@ internal InfiniteScrollList(Lazy<JoinableTaskFactory> joinableTaskFactory)
 
             BindingOperations.EnableCollectionSynchronization(Items, _list.ItemsLock);
 
-            ItemsView = new CollectionViewSource() { Source = Items };
-            ListCollectionView itemsView = ItemsView.View as ListCollectionView;
+            ItemsView = new CollectionViewSource() { Source = Items }.View;
+            ICollectionViewLiveShaping itemsView = (ICollectionViewLiveShaping)ItemsView;
             itemsView.IsLiveFiltering = true;
             itemsView.IsLiveGrouping = true;
             itemsView.LiveFilteringProperties.Add(nameof(PackageItemViewModel.IsPackageVulnerable));
             itemsView.LiveGroupingProperties.Add(nameof(PackageItemViewModel.PackageLevel));
-            itemsView.Filter = item =>
+            ItemsView.Filter = item =>
             {
-                if (item is not null && item.Equals(_loadingVulnerabilitiesStatusIndicator))
-                {
-                    return FilterByVulnerabilities && !(_loadingVulnerabilitiesStatusIndicator.Status == LoadingStatus.NoItemsFound && VulnerablePackagesCount > 0);
-                }
-                PackageItemViewModel vitem = item as PackageItemViewModel;
-                return IsItemInFilter(vitem);
+                return IsItemAcecpted(item);
             };
 
-            DataContext = ItemsListCollection;
+            DataContext = itemsView;
             CheckBoxesEnabled = false;
 
             _loadingStatusIndicator.PropertyChanged += LoadingStatusIndicator_PropertyChanged;
         }
 
+
+        private bool IsItemAcecpted(object item)
+        {
+            if (item is LoadingStatusIndicator)
+            {
+                return IsLoadingIndicatorAccepted(item);
+            }
+
+            if (item is PackageItemViewModel vm)
+            {
+                return IsVulnerablePackageAccepted(vm);
+            }
+
+            return false;
+        }
+
+        private bool IsLoadingIndicatorAccepted(object item)
```

Comment:
There's inconsistencies here. This method accept a parameter of type object, but the caller ensures the object is of type LoadingStatusIndicator. Then we check again if this is not null. This is inconsistent with how  IsVulnerablePackageAccepted is handling casting. Let's decide on one way and stick with it for all filter methods.